pallets/collator-assignment/src/lib.rs

```diff
@@ -287,6 +295,23 @@ pub mod pallet {
                 }
             };
 
+            let mut assigned_containers = new_assigned.container_chains.clone();
+            assigned_containers.retain(|_, v| !v.is_empty());
+
+            // On congestion (not enough collators for all paras) prioritized chains need to pay
+            // the minimum tip of the prioritized chains
+            let maybe_tip: Option<BalanceOf<T>> =
+                if assigned_containers.len() < container_chain_ids.len() + parathreads.len() {
+                    assigned_containers
+                        .into_keys()
+                        .map(T::CollatorAssignmentTip::get_para_tip)
+                        .filter(|o| o.is_some())
```

Comment:
```suggestion
                        .filter_map(T::CollatorAssignmentTip::get_para_tip)
```

Pretty sure there's a clippy lint for this

Reply:
Much better! Clippy didn't warn about this, though https://github.com/moondance-labs/tanssi/actions/runs/8434396122/job/23097544135